subsys/bluetooth/host/hci_core.c

```diff
@@ -1322,6 +1322,12 @@ static void le_legacy_conn_complete(struct net_buf *buf)
 	struct bt_hci_evt_le_conn_complete *evt = (void *)buf->data;
 	struct bt_hci_evt_le_enh_conn_complete enh;
 
+#if defined(CONFIG_BT_BROADCASTER)
+	struct bt_le_ext_adv *adv = bt_le_adv_lookup_legacy();
+
+	(void)bt_le_lim_adv_cancel_timeout(adv);
```

Comment:
This should only be done for connections established in peripheral role.

Enhanced connection complete event could still be delivered without extended advertising support (i.e no advertising set terminated event) so this handling should be moved to the common conn complete function.
This looks like the correct place to move it:
https://github.com/zephyrproject-rtos/zephyr/blob/78540881166a9634667fc646c3f5a685d8960db9/subsys/bluetooth/host/hci_core.c#L1159